index.js

```diff
@@ -1,5 +1,6 @@
 function locate(array, target) {
-  // do work here
+  let flattenArray = [].concat.apply([], array);
+  let flattenMore = [].concat.apply([], flattenArray);
```

Comment:
Although this is pretty interesting how it works, (I had to dig deep to figure out how it worked). It's not very readable. And works only for about 3 levels deep. If the arrays were nested any deeper, this solution wouldn't work.